backend/src/main/java/gov/cdc/usds/simplereport/config/authorization/PermissionHolder.java

```diff
@@ -0,0 +1,27 @@
+package gov.cdc.usds.simplereport.config.authorization;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.HashSet;
+
+/**
+ * Interface for an object that returns a set (usually a singleton) of
+ * organization roles, and a set (usually non-empty) of permissions granted by
+ * those roles.
+ */
+public interface PermissionHolder {
+
+    Set<OrganizationRole> getGrantedRoles();
+
+    default Set<UserPermission> getGrantedPermissions() {
+        Set<OrganizationRole> roles = getGrantedRoles();
+        Set<UserPermission> permissions = new HashSet<>();
+        for (OrganizationRole r : Arrays.asList(OrganizationRole.ADMIN, OrganizationRole.ENTRY_ONLY,
```

Comment:
Hard-coded priority ordering here is annoying—going to fix that in the front-end role label story, and make it more obvious why we're doing it this way.